pkg/kube/kube.go

```diff
@@ -32,11 +44,11 @@ type Kuber interface {
 // Kube contains state for communicating with
 // a kubernetes cluster
 type Kube struct {
-	KubeConfigPath string
-	ClientSet      *kubernetes.Clientset
-	RestConfig     *rest.Config
-	Ctx            context.Context
-	Log            *logrus.Logger
+	Provider   ClientSetProvider
```

Comment:
Whats a Kubernetes client set?

Reply:
Something that knows how to communicate with the Kubernetes API server, generally it also refers to a specific client